maxmin/max3/keep2.go

```diff
@@ -0,0 +1,16 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+func main() {
+	fmt.Println("The program calculates the maximum of three numbers")
+	fmt.Println("Enter three numbers:")
+	var a, b, c float64
+	fmt.Scanln(&a, &b, &c)
+	max2 := (a + b + math.Abs(a-b)) / 2
```

Comment:
You don't really need to introduce new variables since you could override existing values:
```
max := (a + b + math.Abs(a-b)) / 2
max = (max + c + math.Abs(max-c)) / 2